File: 卡码/代码随想录/Java/src/com/kurumi/codeThoughts/二叉树/_02_二叉树属性/LeetCode_0572.java
package com.kurumi.codeThoughts.二叉树._02_二叉树属性;

import com.kurumi.codeThoughts.entity.TreeNode;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 572. 另一棵树的子树
 */
public class LeetCode_0572 {
    static class Solution {
        public boolean compare(TreeNode left, TreeNode right) {
            if (left == null && right != null)
                return false;
            else if (left != null && right == null)
                return false;
            else if (left == null && right == null)
                return true;
            else if (left.val != right.val)
                return false;
            else
                return compare(left.left, right.left) && compare(left.right, right.right);
        }

        public boolean isSubtree(TreeNode root, TreeNode subRoot) {
            if (root == null)
                return false;
            return compare(root, subRoot) || isSubtree(root.left, subRoot) || isSubtree(root.right, subRoot);
        }
    }
}

File: 卡码/代码随想录/1/06-二叉树/Go/02-二叉树属性/LeetCode_0572.go
//572. 另一棵树的子树
package main

import (
	. "binaryTree/types"
)

func compare(left,right *TreeNode) bool{
	if left!=nil&&right==nil {
		return false
	}else if left==nil&&right!=nil {
		return false
	}else if left==nil&&right==nil {
		return true
	}else if left.Val!=right.Val {
		return false
	}else{
		return compare(left.Left,right.Left)&&compare(left.Right,right.Right)
	}
}

func isSubtree(root *TreeNode, subRoot *TreeNode) bool {
	if root==nil {
		return false
	}else{
		return compare(root,subRoot)||isSubtree(root.Left,subRoot)||isSubtree(root.Right,subRoot)
	}
}
